crates/huub/src/constraints/int_diffn.rs

```diff
@@ -0,0 +1,829 @@
+				self.ub_tracker[curr_obj_idx][curr_dimension],
+				sweep[curr_dimension]
+			);
+			let reason = self.explain_propagation(
+				actions,
+				all_fr_explain,
+				fr_support,
+				curr_obj_idx,
+				curr_dimension,
+				false,
+			);
+			actions.set_int_lower_bound(
+				self.box_posn[curr_obj_idx][curr_dimension],
+				sweep[curr_dimension],
+				reason,
+			)?;
+
+			self.lb_tracker[curr_obj_idx][curr_dimension] = sweep[curr_dimension];
+		}
+		Ok(())
+	}
+
+	/// Prune the upper bounds of the domain
+	fn prune_max<P: PropagationActions>(
+		&mut self,
+		actions: &mut P,
+		fr_support: &[usize],
+		curr_obj_idx: usize,
+		curr_dimension: usize,
+		all_fr: &[ForbiddenRegion],
+		all_fr_explain: &[ForbiddenRegion],
+	) -> Result<(), Conflict> {
+		trace!("PRUNE MIN");
+		let mut sweep = vec![];
+		let mut jump = vec![];
+		let mut b = true;
+
+		for i in 0..self.dimensions {
+			sweep.push(self.ub_tracker[curr_obj_idx][i]);
+			jump.push(self.lb_tracker[curr_obj_idx][i] - 1);
+		}
+		let mut infeasible_fr = Self::infeasible_sweep(&sweep, self.dimensions, all_fr);
+		while b && infeasible_fr.is_some() {
+			for j in 0..self.dimensions {
+				jump[j] = cmp::max(jump[j], infeasible_fr.unwrap().lb[j] - 1);
+			}
+			// Contains side-effects to change sweep
+			b = Self::adjust_sweep_max(
+				&mut sweep,
+				&mut jump,
+				&self.lb_tracker[curr_obj_idx],
+				&self.ub_tracker[curr_obj_idx],
+				curr_dimension,
+				self.dimensions,
+			);
+
+			infeasible_fr = Self::infeasible_sweep(&sweep, self.dimensions, all_fr);
+		}
+		if sweep[curr_dimension] != self.ub_tracker[curr_obj_idx][curr_dimension] {
+			trace!(
+				"SET obj {:?} in dimension {} [{:?}..{:?}] < {}",
+				curr_obj_idx,
+				curr_dimension,
+				self.lb_tracker[curr_obj_idx][curr_dimension],
+				self.ub_tracker[curr_obj_idx][curr_dimension],
+				sweep[curr_dimension]
+			);
+			let reason = self.explain_propagation(
+				actions,
+				all_fr_explain,
+				fr_support,
+				curr_obj_idx,
+				curr_dimension,
+				true,
+			);
+
+			actions.set_int_upper_bound(
+				self.box_posn[curr_obj_idx][curr_dimension],
+				sweep[curr_dimension],
+				reason,
+			)?;
+
+			self.ub_tracker[curr_obj_idx][curr_dimension] = sweep[curr_dimension];
+		}
+		Ok(())
+	}
+
+	/// Adjusts the sweep and jump point when pruning the lower bound
+	fn adjust_sweep_min(
+		sweep: &mut [IntVal],
+		jump: &mut [IntVal],
+		curr_obj_lb: &[IntVal],
+		curr_obj_ub: &[IntVal],
+		curr_dimension: usize,
+		dimensions: usize,
+	) -> bool {
+		for i in (0..dimensions).rev() {
+			let rotation = (i + curr_dimension) % dimensions;
+			sweep[rotation] = jump[rotation];
+			jump[rotation] = curr_obj_ub[rotation] + 1;
+			if sweep[rotation] <= curr_obj_ub[rotation] {
+				return true;
+			} else {
+				// Reset sweep-point
+				sweep[rotation] = curr_obj_lb[rotation];
+			}
+		}
+		sweep[curr_dimension] = curr_obj_ub[curr_dimension] + 1;
+		false
+	}
+
+	/// Adjusts the sweep and jump point when pruning the upper bound
```

Comment:
Do we explain somewhere what the sweep and jump point are and what they are used for?